### DISCUSSION TITLE
How can I enable DISK encryption in Orin nx?

### DISCUSSION THREAD

**Discussion by minorukatsuragi** [OE4T/meta-tegra#1421] [General]
Hi.

I have questions about to enable DISK encryption in Orin nx?

 For context, my os is based on yocto kirkstone branch, with l4t  R35.4.1 JetPack release: 5.1.2 
and I am building for nvida jetson Orin NX.

MACHINE ?= "p3768-0000-p3767-0000"

STEP1 Enabled Secure boot

In meta-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles_35.4.1.bb

ODMFUSE_FILE ? = "${S}/bootloader/fuse_config_security.xml"

The image was built with the above settings.

fuse_config_security.xml
--------------------------------------------------------------------------------------------------------------------------------------------
<genericfuse MagicId="0x45535546" version="1.0.0">
    <fuse name="PublicKeyHash" size="64" value="0x9f0ebf0aec1e2bb30c0838096a6d9de5fb86b1277f182acf135b081e345970167a88612b916128984564086129900066255a881948ab83bebf78c7d627f8fe84"/>
    <fuse name="SecureBootKey" size="32" value="0x123456789abcdef0fedcba987654321000112233445566778899aabbccddeeff"/>
    <fuse name="OemK1" size="32" value="0xf3bedbff9cea44c05b08124e8242a71ec1871d55ef4841eb4e59a56b5f88fb2b"/>
    <fuse name="BootSecurityInfo" size="4" value="0x20b"/>
    <fuse name="SecurityMode" size="4" value="0x1"/> </genericfuse>
--------------------------------------------------------------------------------------------------------------------------------------------
Result
The following fuse-related scripts were generated in the generated compressed file.

burnfuses.sh
l4t_sign_image.sh
secureflash.xml
odmfuse_pkc.xml

STEP2
Do you know how can I enable DISK encryption in Orin nx?
I have read the following guide but do not understand how it relates to meta-tegra.
https://docs.nvidia.com/jetson/archives/r35.4.1/DeveloperGuide/text/SD/Security/OpTee.html#sd-security-optee-keymaintenanceandekbs-ekbgeneration
Any information would be very helpful.

Regards.

**Comment by madisongh** [comment]
I don't think we have a reference implementation for L4T R35.x around at the moment, but the basic approach described in [this ELC presentation](https://youtu.be/8VADXvHTx_k) - slides are [here](https://elinux.org/images/1/14/Tales_from_the_crypt_v2.pdf) - could be adapted.
